api/src/wfl/server.clj

```diff
@@ -78,29 +78,23 @@
       wrap-internal-error
       (wrap-json-response {:pretty true})))
 
-(defn ^:private do-update!
-  "Update `_workload` in a database transaction. "
-  [{:keys [id] :as _workload}]
-  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-    (let [workload (workloads/load-workload-for-id tx id)]
+(defn ^:private try-update
+  "Try to update `workload-record` with backstops."
+  [{:keys [id] :as workload-record}]
+  (try
+    (let [workload (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+                     (workloads/load-workload-for-id tx id))]
+      (log/info "Updating workload" :workload (workloads/to-log workload))
       (try
-        (workloads/update-workload! tx workload)
+        (workloads/update-workload! workload)
         (catch UserException e
-          (log/warning "Error updating workload"
+          (log/warning "UserException while updating workload"
                        :workload  (workloads/to-log workload)
                        :exception e)
-          (slack/notify-watchers workload (.getMessage e)))))))
-
-(defn ^:private try-update
-  "Try to update the workflows in `workload` with a backstop."
-  [workload]
-  (try
-    (log/info "Updating workload"
-              :workload (workloads/to-log workload))
-    (do-update! workload)
+          (slack/notify-watchers workload (.getMessage e)))))
     (catch Throwable t
       (log/error "Failed to update workload"
-                 :workload  (workloads/to-log workload)
+                 :workload  (workloads/to-log workload-record)
                  :throwable t))))
```

Comment:
Now the initial load and the update run in separate TXs.
Would it be safer to run the load in the update's transaction?
(Not sure how those overlap in the schema.)

I prefer a separate backstop try to the nesting, but that looks correct.
I like the workload-record distinction.

Reply:
Yeah, good points all around.

I will see about punting the workload load to the update transactions.  Originally I thought I needed the loaded workload to Slack on UserExceptions, but the workload record is sufficient.

I was on the fence about consolidating the backstops anyways (something about try-within-a-try is hard to track) so will un-nest.

Reply:
I can manage try-within-try on a good day, but not every day.

Reply:
Workload loading now takes place in the update's transactions, and try-within-a-try is no more.